pkgs/applications/audio/aaxtomp3/default.nix

```diff
@@ -36,17 +27,51 @@ stdenv.mkDerivation rec {
     hash = "sha256-7a9ZVvobWH/gPxa3cFiPL+vlu8h1Dxtcq0trm3HzlQg=";
   };
 
-  dontBuild = false;
+  # use whitespace to show osh arithmetic is not file redirection
+  patches = [./osh.patch];
 
-  nativeBuildInputs = [ makeWrapper ];
+  buildPhase = ''
+    substituteInPlace AAXtoMP3 \
+      --replace 'GREP="grep"'                   'GREP="${gnugrep}/bin/grep"' \
+      --replace 'GREP="ggrep"'                  'GREP="${gnugrep}/bin/grep"' \
+      --replace 'FIND="find"'                   'FIND="${findutils}/bin/find"' \
+      --replace 'FIND="gfind"'                  'FIND="${findutils}/bin/find"' \
+      --replace 'SED="sed"'                     'SED="${gnused}/bin/sed"' \
+      --replace 'SED="gsed"'                    'SED="${gnused}/bin/sed"' \
+  '';
 
   installPhase = ''
-    install -Dm755 AAXtoMP3 $out/bin/aaxtomp3
-    wrapProgram $out/bin/aaxtomp3 --prefix PATH : ${lib.makeBinPath runtimeInputs}
-    install -Dm755 interactiveAAXtoMP3 $out/bin/interactiveaaxtomp3
-    wrapProgram $out/bin/interactiveaaxtomp3 --prefix PATH : ${lib.makeBinPath runtimeInputs}
+    install -Dm 755 AAXtoMP3 $out/bin/aaxtomp3
+    install -Dm 755 interactiveAAXtoMP3 $out/bin/interactiveaaxtomp3
   '';
 
+  solutions.default = {
+    scripts = [
+      "bin/aaxtomp3"
+      "bin/interactiveaaxtomp3"
+    ];
+    interpreter = "${bash}/bin/bash";
```

Comment:
Not sure if runtimeShell should be used here

Reply:
I have wondered this myself, but my current thinking is: I'm leery of it given the lack of clear documentation on how the runtimeShell attr is set and what it's for. 

I have a hard time convincing myself there are cases where we'd want the shell that interprets most packages to potentially change to a shell it wasn't written for. 

I can imagine choosing a canonical /bin/sh and changing which package it points to, but runtimeShell will be full bash.

I can imagine deciding to change the interpreter scripts internal to nixpkgs (my best guess at what runtimeShell is for) and then fixing whatever is broken by doing so.

Reply:
I have been told not to use `runtimeShell`, when the upstream asks for `bash`; but if that is incorrect, happy to change it.